spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitStreamConnectionDetails.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2012-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.amqp;
+
+import org.springframework.boot.autoconfigure.service.connection.ConnectionDetails;
+
+/**
+ * Details required to establish a connection to a RabbitMQ service.
+ *
+ * @author Eddú Meléndez
+ * @since 3.4.0
+ */
+public interface RabbitStreamConnectionDetails extends ConnectionDetails {
```

Comment:
The `RabbitConnectionDetails` also has these methods:

* `getUsername`
* `getPassword`
* `getVirtualHost`

Which are also contained in `org.springframework.boot.autoconfigure.amqp.RabbitProperties.Stream`. I think we should add these to the `ConnectionDetails` interface.